### ISSUE TITLE
MappingException after upgrade from 0.108.19-SNAPSHOT to 0.111

### ISSUE THREAD

**Issue by shisoft** [mongodb/morphia#748]
Hi. I am trying to upgrade morphia to resolve a performance issue on reference. I have also upgrade mongodb-java-driver to 3.0.0.
Then I am keep getting `org.mongodb.morphia.mapping.MappingException` like

> org.mongodb.morphia.mapping.MappingException: The reference({ "$ref" : "UniversalContact", "$id" : > "50f3f437a6036279edc69c74" }) could not be fetched for net.shisoft.db.obj.User.uc

But both the collection `UniversalContact` and id `50f3f437a6036279edc69c74` exists in the database.

```
@Entity(noClassnameStored = true)
public class User implements java.io.Serializable {
    @Reference
    UniversalContact uc;
    ...
}
```
```
@Entity(noClassnameStored = true)
public class UniversalContact implements java.io.Serializable, Cloneable {
    @Id
    ObjectId id;
    ...
}
```

**Comment by shisoft** [comment]
I've found that this issue was caused because ReferenceMapper tried to fetch the record from another database with the same Morphia instance.

**Comment by shisoft** [comment]
Do I have to use one Morphia instance per datastore?

**Comment by shisoft** [comment]
Now I am using different Morphia instance, but still get the same error

**Comment by shisoft** [comment]
The problem is 
Line 260, `final Datastore ds = mapper.getDatastoreProvider().get();`. In the Mapper.java, Line 108, `private final DatastoreProvider datastoreProvider = new DefaultDatastoreProvider();`, in the  DefaultDatastoreProvider, Line 17 `final Datastore datastore = DatastoreHolder.getInstance().get();`, in the DatastoreHolder, getInstance return  static `INSTANCE`.

Please fix this.

**Comment by evanchooly** [comment]
#729 is likely to have fixed this for you.  There should be a new snapshot available within the hour.  Could you retry with that build and confirm.  https://oss.sonatype.org/content/repositories/snapshots/org/mongodb/morphia/morphia/1.0.0-rc0-SNAPSHOT/

the build timestamp will be later than the 09:30 timestamp there now.

**Comment by evanchooly** [comment]
As far as I can tell, this issue was resolved with #729.  Should you find you still have problems, please post a test case to the [mailing list](https://groups.google.com/forum/#!forum/morphia)
